shapeshift/web
# feat: hide URL params in scanned URL from send/QR form

## Description

Makes the input field cleaner and moves crypto amount logic outside of `src/components/Modals/Send/views/Address` (which should really only handle addresses)

## Pull Request Type


## Issue (if applicable)



closes #5068

## Risk



- High WRT Coinbase wallet URL parsing
- Risks of regressions in account change (though this should theoretically be an improvement here and reset the crypto/fiat amount while switching accounts in the send modals, as the intent was, see the removed TODO)

## Testing



- Regression test QR and send modal sends fully
- Ensure amounts are still populated
- Ensure no parameters are present in the displayed scanned address regardless of whether we're scanning a QR code with/out amount
- Ensure EVM sends still prompt for asset selection
- Ensure Coinbase wallet QR codes show no regressions

### Engineering



### Operations



## Screenshots (if applicable)

### QR Code form

- QR code button - UTXO scanned address i.e no amount (note the 0.1 amount was manually input here)

<img width="541" alt="Screenshot 2023-08-23 at 18 02 22" src="https://github.com/shapeshift/web/assets/17035424/4b58c1f2-9d89-4e21-bc82-d4ebb811b46f">
<img width="541" alt="Screenshot 2023-08-23 at 18 02 22" src="https://github.com/shapeshift/web/assets/17035424/eeb4cce1-4d3d-4870-9df4-7d7efc99b878">
<img width="536" alt="Screenshot 2023-08-23 at 18 02 52" src="https://github.com/shapeshift/web/assets/17035424/332aff93-9b4b-44b8-99a6-c4d2776bac80">

- QR code button - Scanned EVM URL i.e no amount

<img width="548" alt="image" src="https://github.com/shapeshift/web/assets/17035424/365687f3-3c56-43a6-a017-70df55538449">
<img width="546" alt="image" src="https://github.com/shapeshift/web/assets/17035424/33816f88-aefe-4a6a-9796-7f8e09a07e7a">
<img width="549" alt="image" src="https://github.com/shapeshift/web/assets/17035424/03b206e6-35ec-4564-8630-21835b1c2686">

- QR code button - UTXO scanned URL with amount

<img width="557" alt="Screenshot 2023-08-23 at 18 03 12" src="https://github.com/shapeshift/web/assets/17035424/30c11a8f-6d99-4ca3-b3a8-bc348a4eb7e0">
<img width="561" alt="Screenshot 2023-08-23 at 18 03 18" src="https://github.com/shapeshift/web/assets/17035424/02edc330-ecb0-4815-904f-fcf774511d40">
<img width="535" alt="Screenshot 2023-08-23 at 18 03 25" src="https://github.com/shapeshift/web/assets/17035424/4ad72362-ea55-4ee7-871f-0272f85a288f">

- QR code button - EVM scanned URL with amount

<img width="543" alt="image" src="https://github.com/shapeshift/web/assets/17035424/3c03a241-2de6-4b44-8b26-0c011763807a">
<img width="531" alt="image" src="https://github.com/shapeshift/web/assets/17035424/476e99c6-943f-4e6e-b515-91ad12df64c7">
<img width="543" alt="image" src="https://github.com/shapeshift/web/assets/17035424/297845f7-5fe5-4f88-82dd-29334adc37a7">


- Send button - UTXO scanned address i.e no amount (note the 0.1 amount was manually input here)

<img width="552" alt="image" src="https://github.com/shapeshift/web/assets/17035424/3469bf21-2a5e-48c9-b9a8-5234175cac3f">
<img width="536" alt="image" src="https://github.com/shapeshift/web/assets/17035424/185afadb-60fa-41e6-8afa-eb13da12531a">
<img width="554" alt="image" src="https://github.com/shapeshift/web/assets/17035424/836771c9-37a2-4098-b60e-18fb14924c12">

- Send button - UTXO scanned URL with amount

<img width="554" alt="image" src="https://github.com/shapeshift/web/assets/17035424/0d427a88-ae8e-4d98-b66c-48f3370aa843">
<img width="554" alt="image" src="https://github.com/shapeshift/web/assets/17035424/0859e3c1-d83d-4fb6-a833-43b8243f506f">
<img width="575" alt="image" src="https://github.com/shapeshift/web/assets/17035424/281e1a72-ee56-402f-9640-03c4515d95f2">

- Send button - EVM scanned address i.e no amount (note the 0.1 amount was manually input here)

<img width="553" alt="image" src="https://github.com/shapeshift/web/assets/17035424/95ec0c4e-62d2-4456-ac3f-66be5eaa0c09">
<img width="555" alt="image" src="https://github.com/shapeshift/web/assets/17035424/d8108802-047a-4e20-826a-8283640721a1">
<img width="552" alt="image" src="https://github.com/shapeshift/web/assets/17035424/928fa2af-b758-47d1-a8a9-0f84c64efb63">

- Send button - EVM scanned URL with amount

<img width="563" alt="image" src="https://github.com/shapeshift/web/assets/17035424/7b687703-d4a5-413f-bdf8-1fd16cd3b37d">
